gem/lib/pupperware/spec_helper.rb

```diff
@@ -126,6 +126,13 @@ def get_container_name(container)
     inspect_container(container, '{{.Name}}')
   end
 
+  def get_container_hostname(container)
+    hostname = inspect_container(container, '{{.Config.Hostname}}')
+    suffix = inspect_container(container, '{{index .NetworkSettings.Networks.s_default.Aliases 0}}')
```

Comment:
While the right thing ultimately happens, this query is not correct

```
Template parsing error: template: :1:24: executing "" at <.NetworkSettings.Networks.s_default.Aliases>: map has no entry for key "s_default"
queried {{index .NetworkSettings.Networks.s_default.Aliases 0}} of 77ff436b7d11641be9b032978d8a7a95ff7bdae8e9e08df6244534ed0d6ded54: 
running agent puppet_test59 in network pupperware_default against puppet
```

Reply:
Got this sorted by collecting *all* of the aliases with `{{range .NetworkSettings.Networks}}{{.Aliases}}{{end}}` then finding the first one that looks like `foo.bar` in Ruby. We fallback to `.Config.Hostname` when no such alias exists.